File: backend/entity/meal_menu.go
package entity

import (
	"gorm.io/gorm"
)

type MealMenu struct {
	gorm.Model
	MenuType string
	PortionText string

	// MealID ทำหน้าที่เป็น FK
	MealID uint
	Meal Meal `gorm:"foreignKey:MealID"`

	// Menu ทำหน้าที่เป็น FK
	MenuID uint
	Menu Menu `gorm:"foreignKey:MenuID"`

}

File: frontend/src/interfaces/MealMenu.ts
/* import type { MealInterface } from "./Meal";
import type { MenuInterface } from "./Menu"; */
export interface MealMenuInterface {
    ID?: number;
    /* MenuType?: string; */
    PortionText?: string;
    MealID?: number; //FK
    MenuID?: number; //FK
    isFoodItem?: boolean; // เพื่อแยกว่าเป็น FoodItem หรือ Menu
    isSpecialDessert?: boolean; // สำหรับของหวานพิเศษของโรคเบาหวาน
    TargetID?: number; // ใช้เป็น id สำหรับ navigate
    /* Menus?: MenuInterface[];  */
    /* Menu?: MenuInterface; 
    Meals?: MealInterface[];  */               
}